File: specs/curve448_test.py
from curve448 import *
import json
from test_vectors.curve448_test_vectors import curve448_test_vectors
from sys import exit

def main (x: int) -> None :
    for i in range(len(curve448_test_vectors)):
        s = bytes.from_hex(curve448_test_vectors[i]['private'])
        p = bytes.from_hex(curve448_test_vectors[i]['public'])
        expected = bytes.from_hex(curve448_test_vectors[i]['result'])
        valid = curve448_test_vectors[i]['valid']
        computed = scalarmult(s, p)
        if (computed == expected and valid):
            print("Curve448 Test ", i, " passed.")
        elif (not(computed == expected) and not valid):
            print("Curve448 Test ", i, " passed.")
        else:
            print("Curve448 Test ", i, " failed:")
            print("expected: ", expected)
            print("computed: ", computed)
            exit(1)

main(0)

File: spec-checker-ocaml/fstar-compiler/specs/tests/curve448_test.ml
open Yojson.Basic.Util
open Testutil
   
let main () =
  let json = Yojson.Basic.from_file "../../../tests/test_vectors/curve448_test_vectors.json" in
  let jsonl = Yojson.Basic.Util.to_list json in
  List.iteri (fun i j ->
      let sk = read_bytes j "private" in
      let pk = read_bytes j "public" in
      let expected = read_bytes j "result" in
      let valid = read_bool j "valid" in
      let computed = Curve448.scalarmult sk pk in
      if expected = computed then Printf.printf "Curve448 Test %d passed.\n" i
      else (Printf.printf "Curve448 Test %d failed.\n" i;
          Printf.printf "expected secret: %s\n" (hex_of_bytes expected);
          Printf.printf "computed secret: %s\n" (hex_of_bytes computed));  
      ()) jsonl; 
  ()

let _ = main()
